homeassistant/components/upnp/sensor.py

```diff
@@ -99,30 +165,12 @@ class UpnpSensor(UpnpEntity, SensorEntity):
     def __init__(
         self,
         coordinator: UpnpDataUpdateCoordinator,
-        sensor_type: dict[str, str],
+        sensor_entity: UpnpSensorEntityDescription,
     ) -> None:
         """Initialize the base sensor."""
-        super().__init__(coordinator)
-        self._sensor_type = sensor_type
-        self._attr_name = f"{coordinator.device.name} {sensor_type['name']}"
-        self._attr_unique_id = f"{coordinator.device.udn}_{sensor_type['unique_id']}"
-
-    @property
-    def icon(self) -> str:
-        """Icon to use in the frontend, if any."""
-        return "mdi:server-network"
-
-    @property
-    def available(self) -> bool:
-        """Return if entity is available."""
-        return super().available and self.coordinator.data.get(
-            self._sensor_type["device_value_key"]
-        )
-
-    @property
-    def native_unit_of_measurement(self) -> str:
-        """Return the unit of measurement of this entity, if any."""
-        return self._sensor_type["unit"]
+        super().__init__(coordinator, sensor_entity)
+        self._attr_native_unit_of_measurement = sensor_entity.native_unit_of_measurement
```

Comment:
```suggestion
```
This is not needed

Reply:
Same here, would we not need to set the unit of measurement?

Reply:
It's set by the entity description